lib/Sema/ConstraintSystem.cpp

```diff
@@ -2620,10 +2619,47 @@ ConstraintSystem::getTypeOfMemberReference(
     auto *functionType = fullFunctionType->getResult()->getAs<FunctionType>();
     functionType = unwrapPropertyWrapperParameterTypes(*this, funcDecl, functionRefKind,
                                                        functionType, locator);
+    auto &ctx = DC->getASTContext();
+    auto *parentModule = useDC->getParentModule();
+    bool baseSendable = isSendableType(parentModule, baseOpenedTy);
+    bool inferredSendable =
+        ctx.LangOpts.hasFeature(Feature::InferSendableMethods);
+
+    if (inferredSendable) {
+      auto sendableProtocol = parentModule->getASTContext().getProtocol(
+          KnownProtocolKind::Sendable);
+      auto baseConformance = TypeChecker::conformsToProtocol(
+          baseOpenedTy, sendableProtocol, parentModule);
+
+      if (baseSendable) {
+        // Add @Sendable to functions without conditional conformances
+        if (baseConformance.getConditionalRequirements().empty())
+          functionType = functionType->withExtInfo(functionType->getExtInfo().withConcurrent())->getAs<FunctionType>();
+
+        // Handle Conditional Conformances
+        for (auto req : baseConformance.getConditionalRequirements()) {
+          if (!TypeChecker::conformsToProtocol(req.getFirstType(),
+                                               sendableProtocol, parentModule)
```

Comment:
Hmm, this is assuming that the conditional requirement is a `Sendable` conformance on the subject type. Can we instead use `checkGenericArguments`? We should be able to gather the right substitutions from the base type since we know it's fully resolved at this point (or resolved-enough to check these conditional requirements).

Reply:
What should be in the substitution map? Should it include all the conformance requirements or just sendable?